rmw_fastrtps_dynamic_cpp/src/rmw_serialize.cpp

```diff
@@ -45,6 +45,7 @@ rmw_serialize(
   auto data_length = tss->getEstimatedSerializedSize(ros_message, ts->data);
   if (serialized_message->buffer_capacity < data_length) {
     if (rmw_serialized_message_resize(serialized_message, data_length) != RMW_RET_OK) {
+      rmw_reset_error();
       RMW_SET_ERROR_MSG("unable to dynamically resize serialized message");
```

Comment:
or maybe we can do the following?

```suggestion
      RMW_SET_ERROR_MSG_AND_APPEND_PREV_ERROR("unable to dynamically resize serialized message");
```

Reply:
In this case, I'm not sure it is a good idea.  `rmw_serialized_message_resize` is defined to be the same as `rcutils_uint8_array_resize`.  At this level, getting an error message that a `uint8_array` resize failed is just confusing, in my opinion.  So in this case, I think it is better to clear out the previous error and set the new one instead.